File: 프로그래머스/0/120862. 최댓값 만들기 （2）/최댓값 만들기 （2）.js
function solution(numbers) {
    // 정수형에서 표현 가능한 가장 작은 안전 값 -9007199254740991
    // 맨 처음 최댓값 중 음수가 있을 수 있으므로 사용
    var answer = Number.MIN_SAFE_INTEGER;
    
    for (let i = 0; i < numbers.length; i++) {
        for (let j = i + 1; j < numbers.length; j++) {
            // 다음 인덱스와 곱하기, 가장 큰 수 반환
            answer = Math.max(answer, numbers[i] * numbers[j]);
        }
    }
    
    return answer;
}

File: 프로그래머스/0/120862. 최댓값 만들기 （2）/최댓값 만들기 （2）.java
class Solution {
    public int solution(int[] numbers) {
        // 정수의 최소값은 -231(-2,147,483,648)
        // 맨 처음 최댓값 중 음수가 있을 수 있으므로 사용
        int answer = Integer.MIN_VALUE;
        int max = 0;
        
        for (int i = 0; i < numbers.length; i++) {
            for (int j = i + 1; j < numbers.length; j++) {
                // 다음 인덱스와 곱하기
                max = numbers[i] * numbers[j];
                
                if (answer <= max) { // 현재 최댓값보다 크면 변경
                    answer = max;
                }
            }
        }
        
        return answer;
    }
}